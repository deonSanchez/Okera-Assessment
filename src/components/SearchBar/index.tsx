import { useState } from 'react';
import { useNavigate, createSearchParams } from 'react-router-dom';

import { TextField, Grid, Button } from '@mui/material';
import useStyles from './styles';

const SearchBar = (): JSX.Element => {
  const classes = useStyles();
  const [searchValue, setSearchValue] = useState('');
  let navigate = useNavigate();

  const handleSearch = () => {
    navigate({
      pathname: `/character`,
      search: createSearchParams({ name: searchValue }).toString(),
    });
  };

  const handleKeyDown = (event) => {
    if (event.keyCode === 13) {
      handleSearch();
    }
  };

  return (
    <Grid container justifyContent="flex-start" alignItems="center" spacing={3}>
      <Grid item xs>
        <TextField
          fullWidth
          placeholder="Search Character Name"
          onChange={(e) => setSearchValue(e.target.value)}
          onKeyDown={(e) => handleKeyDown(e)}
          size="small"
          value={searchValue}
        />
      </Grid>
      <Grid item>
        <Button
          className={classes.button}
          onClick={handleSearch}
          variant="contained"
          size="large"
        >
          Search Character
        </Button>
      </Grid>
    </Grid>
  );
};

export default SearchBar;
