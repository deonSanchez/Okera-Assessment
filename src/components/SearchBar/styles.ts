import { makeStyles } from '@mui/styles';

const useStyles = makeStyles(() => ({
  button: {
    textTransform: 'none',
  },
}));

export default useStyles;
