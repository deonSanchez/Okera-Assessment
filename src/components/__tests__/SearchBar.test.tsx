import { render } from '@testing-library/react';
import SearchBar from 'components/SearchBar';
import { MemoryRouter } from 'react-router-dom';

describe('SearchBar', () => {
  test('should render SearchBar', () => {
    const { container } = render(<SearchBar />, { wrapper: MemoryRouter });

    expect(container).toMatchSnapshot();
  });
});
