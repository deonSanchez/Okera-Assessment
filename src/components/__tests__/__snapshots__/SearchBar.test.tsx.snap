// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render SearchBar 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-1pnytec-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            placeholder="Search Character Name"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-hdw1oc"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root makeStyles-button-1 css-jh47zj-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Search Character
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
